bundle/src/main/java/com/adobe/acs/commons/dam/impl/AssetsFolderPropertiesSupport.java

```diff
@@ -0,0 +1,247 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+
+@Component(
+        label = "ACS AEM Commons - Assets Folder Properties Support",
+        policy = ConfigurationPolicy.REQUIRE,
+        immediate = true
+)
+@Properties({
+        @Property(
+                name = "service.ranking",
+                intValue = -2000,
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.filter.scope",
+                value = "REQUEST",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.filter.pattern",
+                value = "/content/dam/.*",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.methods",
+                value = "GET",
+                propertyPrivate = true
+        ),
+        @Property(
+                name = "sling.servlet.resourceTypes",
+                value = "acs-commons/touchui-widgets/asset-folder-properties-support",
+                propertyPrivate = true
+        )
+})
+@Service
+public class AssetsFolderPropertiesSupport extends SlingSafeMethodsServlet implements Filter, SlingPostProcessor {
+    private static final Logger log = LoggerFactory.getLogger(AssetsFolderPropertiesSupport.class);
+
+    private static final String DAM_PATH_PREFIX = "/content/dam";
+    private static final String POST_METHOD = "post";
+    private static final String OPERATION = ":operation";
+    private static final String DAM_FOLDER_SHARE_OPERATION = "dam.share.folder";
+    private static final String GRANITE_UI_FORM_VALUES = "granite.ui.form.values";
+
+    /**
+     * This method is responsible for post processing POSTs to the FolderShareHandler PostOperation (:operation = dam.share.folder).
+     * This method will store a whitelisted set of request parameters to their relative location off of the [sling:*Folder] node.
+     *
+     * Note, this is executed AFTER the OOTB FolderShareHandler PostOperation.
+     *
+     * At this time this method only supports single-value Strings and ignores all @typeHints.
+     *
+     * This method must fail fast via the accepts(...) method.
+     *
+     * @param servletRequest the request object
+     * @param servletResponse the response object
+     * @param chain the filter chain
+     * @throws IOException
+     * @throws ServletException
+     */
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
+        final SlingHttpServletRequest request = (SlingHttpServletRequest) servletRequest;
+        final SlingHttpServletResponse response = (SlingHttpServletResponse) servletResponse;
+
+        if (!accepts(request)) {
+            chain.doFilter(request, response);
+            return;
```

Comment:
ill leave this one; oddly(?) i find nested if/else statements to be less comprehendible (especially as depth increases) -- i think because having a flat-linear decision spaces reduces the amount of state i have to remember the further down the function i go.